programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -654,49 +645,55 @@ static void process_ipc_command(uv_stream_t *s, ssize_t len, char* base) {
 
 
 static void on_cmd(uv_stream_t *s, ssize_t len, const uv_buf_t *b) {
+    struct ipc_conn_s *ipc_client = (struct ipc_conn_s*)s;
     if (len < 0) {
-        ZITI_LOG(WARN, "received from client - %s. Closing connection.", uv_err_name(len));
-        struct ipc_conn_s *del_ipc_client = NULL;
-        LIST_FOREACH(del_ipc_client, &ipc_clients_list, _next_ipc_cmd) {
-            if((uv_stream_t *)del_ipc_client->ipc_client_conn == s) {
-                break;
-            }
+        if (len != UV_EOF) {
+            ZITI_LOG(WARN, "received from client - %s. Closing connection.", uv_err_name(len));
         }
-        if (del_ipc_client) {
-            LIST_REMOVE(del_ipc_client, _next_ipc_cmd);
-            free(del_ipc_client);
-        }
-        uv_close((uv_handle_t *) s, (uv_close_cb) free);
-        ZITI_LOG(WARN,"IPC client connection closed, count: %d", sizeof_ipc_clients_list());
+
+        LIST_REMOVE(ipc_client, _next_ipc_cmd);
+
+        json_tokener *tokener = s->data;
+        if (tokener) json_tokener_free(tokener);
+        uv_close((uv_handle_t *) &ipc_client->ipc, (uv_close_cb) free);
+        ZITI_LOG(DEBUG, "IPC client connection closed, count: %d", sizeof_ipc_clients_list());
 
     } else {
-        ZITI_LOG(INFO, "received cmd <%.*s>", (int) len, b->base);
+        ZITI_LOG(DEBUG, "received cmd <%.*s>", (int) len, b->base);
 
-#if LAST_CHAR_IPC_CMD == '\0'
-        process_ipc_command(s, len, b->base);
-#else
-        queue_ipc_command(ipc_cmd_ctx, len, b->base);
-        if (b->base[len-1] == LAST_CHAR_IPC_CMD) {
-            char* new_buff = ipc_cmd_buffered(ipc_cmd_ctx);
-            ZITI_LOG(TRACE, "buffered cmd <%.*s>", (int) (strlen(new_buff) + 1), new_buff);
-            process_ipc_command(s, strlen(new_buff) + 1, new_buff);
-            free(new_buff);
+        json_tokener *parser = s->data;
+
+        size_t processed = 0;
+        while (processed < len) {
```

Comment:
Shouldn't this have some kind of escape-hatch max value? Something like a 1mb or 10 mb? something 'crazy' but not too crazy?

Reply:
previous implementation was not correct and would definitely crash if you send over 16K of data